include/EVT/dev/encoder.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef _EVT_ENCODER_
+#define _EVT_ENCODER_
+
+#include <EVT/io/GPIO.hpp>
+
+namespace EVT::core::DEV {
+class Encoder {
+public:
+    /**
+     * Constructs an Encoder instance. Call update() within the main loop to update encoder values.
+     * @param a a pin of the encoder
+     * @param b b pin of the encoder
+     * @param range range of the encoder positions (should be positive)
+     * @param initialPosition initial position the encoder is in
+     */
+    Encoder(IO::GPIO& a, IO::GPIO& b, int64_t range, int64_t initialPosition);
+
+    /**
+     * An Enum that represents the direction the encoder is being spun in.
+     * The direction should reset to Static after a couple of reads with no changes.
+     */
+    enum Direction {
+        Left = -1,
+        Static = 0,
+        Right = 1
+    };
+
+    /**
+     * Reads and updates the encoder rotation and the absolute position.
+     * @return what position was changed by, should be in this range: [-3,3]
```

Comment:
What use case does this have?

Reply:
If the value gets capped, you can know how much they over-ran the value.

Reply:
I removed this because it was kind of unnecessary, in retrospect.